tools/visual-pipeline-and-platform-evaluation-tool/vippet/pipeline_runner.py

```diff
@@ -156,6 +173,31 @@ def run(self, pipeline_command: str, total_streams: int) -> PipelineRunResult:
                         exit_code = process.wait()
                         break
 
+                # If there was no activity for a prolonged period, treat as hang
+                if (
+                    not self.cancelled
+                    and (time.time() - last_activity_time) > self.inactivity_timeout
+                ):
+                    self.logger.error(
+                        "No new logs on stdout/stderr for %s seconds; "
+                        "terminating pipeline as potentially hung",
+                        self.inactivity_timeout,
+                    )
+                    process.terminate()
+                    try:
+                        exit_code = process.wait(timeout=5)
+                    except Exception:
+                        self.logger.warning(
+                            "Process did not terminate gracefully after inactivity; killing it"
+                        )
+                        process.kill()
+                        exit_code = process.wait()
+
+                    raise RuntimeError(
+                        f"Pipeline execution terminated due to inactivity timeout "
+                        f"({self.inactivity_timeout} seconds without stdout/stderr logs)."
+                    )
```

Comment:
`exit_code` is not used in this branch so it can be removed

```suggestion
                    try:
                        process.wait(timeout=5)
                    except Exception:
                        self.logger.warning(
                            "Process did not terminate gracefully after inactivity; killing it"
                        )
                        process.kill()
                        process.wait()

                    raise RuntimeError(
                        f"Pipeline execution terminated due to inactivity timeout "
                        f"({self.inactivity_timeout} seconds without stdout/stderr logs)."
                    )
```